deploy/terraform/terraform-units/modules/sap_deployer/variables_local.tf

```
@@ -3,84 +3,49 @@ Description:
 
   Define local variables.
 */
-
-variable "region_mapping" {
-  type        = map(string)
-  description = "Region Mapping: Full = Single CHAR, 4-CHAR"
-
-  # 28 Regions 
-
-  default = {
-    westus             = "weus"
-    westus2            = "wus2"
-    centralus          = "ceus"
-    eastus             = "eaus"
-    eastus2            = "eus2"
-    northcentralus     = "ncus"
-    southcentralus     = "scus"
-    westcentralus      = "wcus"
-    northeurope        = "noeu"
-    westeurope         = "weeu"
-    eastasia           = "eaas"
-    southeastasia      = "seas"
-    brazilsouth        = "brso"
-    japaneast          = "jpea"
-    japanwest          = "jpwe"
-    centralindia       = "cein"
-    southindia         = "soin"
-    westindia          = "wein"
-    uksouth2           = "uks2"
-    uknorth            = "ukno"
-    canadacentral      = "cace"
-    canadaeast         = "caea"
-    australiaeast      = "auea"
-    australiasoutheast = "ause"
-    uksouth            = "ukso"
-    ukwest             = "ukwe"
-    koreacentral       = "koce"
-    koreasouth         = "koso"
-  }
+variable naming {
+  description = "naming convention"
 }
 
 // Set defaults
 locals {
 
-  // Post fix for all deployed resources
-  postfix = random_id.deployer.hex
+  storageaccount_names = var.naming.storageaccount_names.DEPLOYER
+  virtualmachine_names = var.naming.virtualmachine_names.DEPLOYER
+  keyvault_names       = var.naming.keyvault_names.DEPLOYER
+  resource_suffixes    = var.naming.resource_suffixes
 
   // Default option(s):
   enable_secure_transfer = try(var.options.enable_secure_transfer, true)
 
   // Resource group and location
 
   region             = try(var.infrastructure.region, "")
-  environment        = try(var.infrastructure.environment, "")
-  location_short     = try(var.region_mapping[local.region], "unkn")
-  vnet_mgmt_tempname = local.vnet_mgmt.name
-  prefix             = try(var.infrastructure.resource_group.name, upper(format("%s-%s-%s", local.environment, local.location_short, substr(local.vnet_mgmt_tempname, 0, 7))))
-  sa_prefix          = lower(format("%s%s%sdiag", substr(local.environment, 0, 5), local.location_short, substr(local.vnet_mgmt_tempname, 0, 7)))
-  rg_name            = try(var.infrastructure.resource_group.name, format("%s-INFRASTRUCTURE", local.prefix))
+  vnet_mgmt_tempname = try(local.vnet_mgmt.name, "")
+  prefix             = try(var.infrastructure.resource_group.name, var.naming.prefix.DEPLOYER)
+  rg_name            = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes["deployer-rg"]))
```

Step into the dict the same way :) 
Same applies to other places - I will not mark all of them.
```suggestion
  rg_name            = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes.deployer-rg))
```